Clarify the bang-ness of to_unit_name

thegedge opened the pull request:
Previously we had `to_unit_name`, but this method could raise, so it should have a bang suffix.

I've resolved this by having both `to_unit_name` and `to_unit_name!`, and using the `raise`-ing nature of `to_unit_name!` to simplify `Measurable#initialize`.

kmcphillips commented on lib/measured/unit_system.rb:
```diff
@@ -25,6 +25,10 @@ def unit?(name)
   end
 
   def to_unit_name(name)
+    to_unit_name!(name) rescue nil
```
This should rescue the specific error class that we raise. If there's a `nil` error or some other exception we don't expect we shouldn't silently swallow it. Always be explicit about the exception types we rescue.

thegedge replied:
Ah yes, good catch!

thegedge replied:
Fixed

kmcphillips commented:
Haha... @imran-iq I struggled so hard to create a system of units and names for things that don't actually exist.

imran-iq commented on test/case_insensitive_unit_system_test.rb:
```diff
@@ -55,9 +55,25 @@ class Measured::CaseInsensitiveUnitSystemTest < ActiveSupport::TestCase
+  test "#to_unit_name! raises if not found" do
     assert_raises Measured::UnitError do
-      Magic.unit_system.to_unit_name("thunder")
+      Magic.unit_system.to_unit_name!("thunder")
```
Maybe test error message too? `assert_raises_with_message Measured::UnitError, "Unit 'thunder' does not exist" do`

imran-iq commented on test/unit_system_test.rb:
```diff
@@ -56,9 +56,25 @@ class Measured::UnitSystemTest < ActiveSupport::TestCase
+  test "#to_unit_name! raises if not found" do
     assert_raises Measured::UnitError do
-      CaseSensitiveMagic.unit_system.to_unit_name("thunder")
+      CaseSensitiveMagic.unit_system.to_unit_name!("thunder")
```
Same as above
`assert_raises_with_message Measured::UnitError, "Unit 'thunder' does not exist" do`

thegedge commented:
Added `assert_raises_with_message` to our test helper and updated the two tests. I'd actually like to add some new exception types to be more explicit (I'll cut an issue for that).
